### ISSUE TITLE
Enhance filters.assign

### ISSUE THREAD

**Issue by abellgithub** [PDAL/PDAL#1468]
filters.assign could do more than a single blanket value assignment quite easily.

**Comment by hobu** [comment]
> do more than a single blanket value assignment quite easily.

Can you provide more detail on this?

**Comment by abellgithub** [comment]
We discussed allowing setting values based on ranges and such.

**Comment by chambbj** [comment]
It would be nice if we could have a one liner assignment that includes a WHERE clause for conditional assignment. It would also be nice if we could move to more familiar operators for comparison and assignment.

For example,

```
{
  "type": "filters.assign",
  "assignment": "Classification = 0 WHERE HeightAboveGround > 1.0"
}
```

Furthermore, boolean operators for the WHERE clause should be supported.

```
{
  "type": "filters.assign",
  "assignment": "UserData = 1 WHERE (Red > 100 AND Green < 50)"
}
```

Perhaps as an offshoot of this feature request, all filters could inherit a WHERE argument that limits the data that they shall process.

In all cases, the ability to subset the data that will be processed by a stage, while preserving that data that is not processed, eliminates many (most?) needs to use tags and inputs, thus creating multiple passes through the pipeline.

**Comment by hobu** [comment]
#2638 also

**Comment by lowbrows** [comment]
Hello,
Looking at Enhance filters.assign #1468

Tried this assign stage directly after running the HAG_DEM and got :
PDAL: filters.assign: Invalid value 'Classification = 7 WHERE (HeightAboveGround > 100.0 OR HeightAboveGround < -20.0)' for argument 'assignment'.

Tried the syntax given in Improve expression support in filters.assign #3202 and got a Parser ! error.

Just wondering if I need to take the long way around with tags or does someone have a this down.

Thanks!

What am I doing wrong here?

**Comment by Vicent-Ribas** [comment]
Hi, I have encountered the same issue, I am trying to filter.assign points within 1 and 3 m above ground with a specific clasification value, and when I write `value="Classification = 3 WHERE HeightAboveGround > 1 AND HeightAboveGround < 3" ` I get RuntimeError: filters.assign: Invalid value for argument 'value': Found 'AND' after valid expression. 

Not working either with brackets before the expression - `value="Classification = 3 WHERE (HeightAboveGround > 1 AND HeightAboveGround < 3)"`. 

Just wondering if there is a way to do this. Thanks!
